File: skellig-test-step-reader-sts/src/main/java/org/skellig/teststep/reader/sts/StsTestStepReader.java
package org.skellig.teststep.reader.sts;

import org.skellig.teststep.reader.TestStepReader;

import java.nio.file.Path;
import java.util.List;
import java.util.Map;

public class StsTestStepReader implements TestStepReader {

    private StsFileParser parser;

    public StsTestStepReader() {
        this.parser = new StsFileParser();
    }

    @Override
    public List<Map<String, Object>> read(Path fileName) {
        return parser.parse(fileName);
    }
}

File: skellig-test-step-reader-sts/src/main/kotlin/org/skellig/teststep/reader/sts/StsTestStepReader.kt
package org.skellig.teststep.reader.sts

import org.skellig.teststep.reader.TestStepReader
import java.io.InputStream
import java.nio.file.Path

class StsTestStepReader : TestStepReader {

    private var parser: StsFileParser = StsFileParser()

    override fun read(inputStream: InputStream): List<Map<String, Any?>> {
        return parser.parse(inputStream)
    }
}